src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return Storage.addManufacturer(manufacturer);
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(m -> m.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer manufacturerToUpdate = Storage.manufacturers.stream()
+                .filter(m -> m.getId().equals(manufacturer.getId()))
+                .findFirst().get();
+        manufacturerToUpdate.setCountry(manufacturer.getCountry());
+        manufacturerToUpdate.setName(manufacturer.getName());
```

Comment:
this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

Reply:
done